### PR TITLE
Rewrited sign up controller

### PR DESCRIPTION
**Pull request by RomanDeveloperAcc**
Rewrited sign up controller:
- changed private fields to public, where private fields are not necessary;
- removed .bind() methods;
- reordered imports and improved paths;
- changed .forEach() methods to .map();
- fixed try catch and error handling logic;

### CODE REVIEW

**Comment by mplesha** [comment]
confilcts

**Comment by RomanDeveloperAcc** [comment]
> confilcts

resolved

File: src/app/component/auth/components/sign-up/sign-up.component.ts [hunk 1]
```diff
@@ -124,51 +126,52 @@ export class SignUpComponent implements OnInit {
...
-    try {
-      errors.error.forEach(error => {
-        if (error.name === 'email') {
-          this.emailErrorMessageBackEnd = error.message;
-        } else if (error.name === 'password') {
-          this.passwordErrorMessageBackEnd = error.message;
-        }
-      });
-    } catch (e) {
+    if (!errors.error[0]) {
```
Comment (mplesha) [inline src/app/component/auth/components/sign-up/sign-up.component.ts:129]: change it - your code will fail on an empty array

File: src/app/component/auth/components/sign-up/sign-up.component.ts [hunk 2]
```diff
@@ -87,7 +86,7 @@ export class SignUpComponent implements OnInit {
   }
 
   private onSubmitError(errors: HttpErrorResponse): void {
-    errors.error.forEach(error => {
+    errors.error.map(error => {
       switch (error.name) {
```
Comment (mplesha) [inline src/app/component/auth/components/sign-up/sign-up.component.ts:90]: very strange switch - it has only one case. should be changed

File: src/app/component/auth/components/sign-up/sign-up.component.ts [hunk 3]
```diff
@@ -52,13 +50,14 @@ export class SignUpComponent implements OnInit {
...
     this.setNullAllMessage();
     this.loadingAnim = true;
     this.userOwnSecurityService.signUp(userOwnRegister)
-      .subscribe({
-        next:  this.onSubmitSuccess.bind(this),
-        error: this.onSubmitError.bind(this)
+      .subscribe((data) => {
+        this.onSubmitSuccess(data);
+      }, (error) => {
+        this.onSubmitError(error);
```
Comment (mplesha) [inline src/app/component/auth/components/sign-up/sign-up.component.ts:60]:
```js
.subscribe((data) => this.onSubmitSuccess(data), (error) => this.onSubmitError(error))
```
If you have only one short line in a body of an arrow function - you shouldn't write it from new line. A single-lined arrow function looks better. Excepting `no-confusing-arrow`

Comment (RomanDeveloperAcc) [reply src/app/component/auth/components/sign-up/sign-up.component.ts]: > ```js
> .subscribe((data) => this.onSubmitSuccess(data), (error) => this.onSubmitError(error))
> ```
> 
> If you have only one short line in a body of an arrow function - you shouldn't write it from new line. A single-lined arrow function looks better. Excepting `no-confusing-arrow`
> 
> Change here and anywhere.

Fixed

File: src/app/component/auth/components/sign-up/sign-up.component.ts (hunk 2 above)
Comment (RomanDeveloperAcc) [reply src/app/component/auth/components/sign-up/sign-up.component.ts]: > very strange switch - it has only one case. should be changed

Fixed

File: src/app/component/auth/components/sign-up/sign-up.component.ts (hunk 1 above)
Comment (RomanDeveloperAcc) [reply src/app/component/auth/components/sign-up/sign-up.component.ts]: > change it - your code will fail on an empty array

Fixed

**Pull request merged**